WordPress/Classes/Utility/PushNotificationsManager.swift

```diff
@@ -53,18 +53,31 @@ final public class PushNotificationsManager: NSObject {
         return sharedApplication.applicationState
     }
 
+    private var didRegisterForRemoteNotifications = false
 
+
+    /// Enables or disables remote notifications based on current settings.
     /// Registers the device for Remote Notifications: Badge + Sounds + Alerts
     ///
-    @objc func registerForRemoteNotifications() {
+    @objc func setupRemoteNotifications() {
         guard JetpackNotificationMigrationService.shared.shouldPresentNotifications() else {
+            disableRemoteNotifications()
             return
         }
 
         sharedApplication.registerForRemoteNotifications()
+        didRegisterForRemoteNotifications = true
+        return
     }
 
-
+    private func disableRemoteNotifications() {
+        if !didRegisterForRemoteNotifications {
+            sharedApplication.registerForRemoteNotifications()
+        }
```

Comment:
Turns out, calling `unregisterForRemoteNotifications()` while `registerForRemoteNotifications()` has not been called during this launch does nothing.

Without this, disabling notifications on launch doesn't properly work.